compiler/rustc_typeck/src/astconv/mod.rs

```diff
@@ -414,10 +414,25 @@ impl<'o, 'tcx> dyn AstConv<'tcx> + 'o {
                             tcx.ty_error().into()
                         }
                     }
-                    GenericParamDefKind::Const => {
+                    GenericParamDefKind::Const { has_default } => {
                         let ty = tcx.at(span).type_of(param.def_id);
-                        // FIXME(const_generics:defaults)
-                        if infer_args {
+                        if !infer_args && has_default {
+                            if default_needs_object_self(param) {
+                                missing_type_params.push(param.name.to_string());
+                                tcx.ty_error().into()
+                            } else {
+                                // This is a default type parameter.
+                                self.normalize_ty(
+                                    span,
+                                    tcx.at(span).type_of(param.def_id).subst_spanned(
+                                        tcx,
+                                        substs.unwrap(),
+                                        Some(span),
+                                    ),
+                                )
+                                .into()
+                            }
```

Comment:
you are returning `GenericArg::Type` here, you can use `tcx.const_error().into()` instead.

To get the actual gefault value we probably have to store it somewhere.

Reply:
To get the correct default value I think we want to use
```rust
ty::Const::from_anon_const(
                                == GenericArg::from(
                                self.tcx(),
                                    self.tcx().type_of(param.def_id).subst(self.tcx(), substs),
                                param.def_id.expect_local(),
                                )
                            ))
```

Reply:
I changed this as well, not sure if it's correct either as I had to change this code block quite a bit